lib/Parse/ParseDecl.cpp

```diff
@@ -916,7 +916,7 @@ bool Parser::parseSpecializeAttributeArguments(
         }
       }
       if (ParamLabel == "spi") {
-        if (!Tok.is(tok::identifier)) {
+        if (Tok.getText() != "_" && !Tok.is(tok::identifier)) {
```

Comment:
How about `!Tok.is(tok::kw__)`? Or even this:
```suggestion
        if (Tok.isIdentifierOrUnderscore()) {
```

Reply:
gtk!